### PR TITLE
Close sockets immediately

### PR DESCRIPTION
**Pull request by ulex**
Motivation: most of the SocketWire tests are completed not due a proper interruption, but after a some magic timeout (you can observe it in test execution time and logs)
These delays became noticeable in the new unit test runner.

### CODE REVIEW

File: rd-net/RdFramework/Impl/SocketWire.cs [hunk 1]
```diff
@@ -376,14 +376,21 @@ protected void AddTerminationActions([NotNull] Thread receiverThread)
               Monitor.PulseAll(Lock);
             }
 
+            CloseServerSocket();
```
Comment (knah) [inline rd-net/RdFramework/Impl/SocketWire.cs]: Server socket is already closed on server lifetime termination on line 502 (old) / 513 (new).
However, due to LIFO termination order of lifetimes, AddTerminationAction (and Thread.Join) are executed before socket-closing action from CreateServerSocket.
A possibly better approach would be to move all thread-joining actions later in termination order by adding the to a child LifetimeDef added to wire lifetime early in wire lifecycle. It's up for discussion though.

Comment (ulex) [reply rd-net/RdFramework/Impl/SocketWire.cs]: Nice catch, thanks. Didn't expect to find CloseSocket there. Introducing a child lifetime seems like over engineering to me. What about a contract of ownership transfer when SocketWire is created? I updated the review.

File: rd-net/RdFramework/Impl/SocketWire.cs [hunk 2]
```diff
@@ -617,7 +598,7 @@ public ServerFactory(Lifetime lifetime, IScheduler scheduler, IPEndPoint endpoin
         IPEndPoint endpoint = null
       )
       {
-        var serverSocket = Server.CreateServerSocket(lifetime, endpoint);
+        var serverSocket = Server.CreateServerSocket(endpoint);
```
Comment (knah) [inline rd-net/RdFramework/Impl/SocketWire.cs]: ServerFactory's serverSocket lifetime should be greater than lifetimes of individual Server instances it creates. As this is written right now, ServerFactory's socket will be closed after the first Server disconnects, which is not the desired behavior.

Comment (ulex) [reply rd-net/RdFramework/Impl/SocketWire.cs]: It does not work as you described (lifetime is common), but I have pushed theoretical fix.

**Pull request merged**